Makefile

```diff
@@ -29,112 +29,86 @@
 ###############################################################################
 include config.mk
 
-.PHONY : all
-.PHONY : install-dev
 .PHONY : virtualenv
-.PHONY : install-workflow-deps
 .PHONY : install-merlin
-.PHONY : clean-output
-.PHONY : clean-docs
-.PHONY : clean-release
-.PHONY : clean-py
-.PHONY : clean
-.PHONY : release
+.PHONY : install-workflow-deps
+.PHONY : install-merlin-dev
 .PHONY : unit-tests
-.PHONY : cli-tests
+.PHONY : e2e-tests
 .PHONY : tests
 .PHONY : fix-style
 .PHONY : check-style
 .PHONY : check-camel-case
 .PHONY : checks
 .PHONY : reqlist
-.PHONY : check-variables
-
-
-all: install-dev install-merlin install-workflow-deps
-
-
-# install requirements
-install-dev: virtualenv
-	$(PIP) install -r requirements/dev.txt
-
-
-check-variables:
-	- echo MAX_LINE_LENGTH $(MAX_LINE_LENGTH)
-
+.PHONY : release
+.PHONY : clean-release
+.PHONY : clean-output
+.PHONY : clean-docs
+.PHONY : clean-py
+.PHONY : clean
 
-# this only works outside the venv
+# this only works outside the venv - if run from inside a custom venv, or any target that depends on this,
+# you will break your venv.
 virtualenv:
-	$(PYTHON) -m venv $(VENV) --prompt $(PENV) --system-site-packages
-	$(PIP) install --upgrade pip
-
+	(
+	    $(PYTHON) -m venv $(VENV) --prompt $(PENV) --system-site-packages; \
+	    . $(VENV)/bin/activate; \
```

Comment:
this might not work in tcsh

Reply:
i actually don't think you need to 'activate' the venv, b/c the $(PIP) string points to the pip inside that venv already

Reply:
oh, I didn't notice that! I'll remove them - this fell out of automating testing, bc those targets fail without activating it. I don't remember what prompted the thought, but the concern was that running the install targets without activating (pathing pip to the venv) was just installing to the environment if we didn't guaranteeing the venv, and would then fail. Should've looked at the config vars more closely.

Reply:
yeah i think it's the config variables that you can use to change this behavior if needed